src/core.js

```diff
@@ -201,6 +201,24 @@ export default class Base extends EventEmitter {
   }
 
   runHook(str, m, ...args) {
+    const publicHooks = l.hooks(m).toJS();
+
+    if (l.validPublicHooks.indexOf(str) !== -1) {
+      // If the SDK has been configured with a hook handler, run it.
+      if (typeof publicHooks[str] === 'function') {
+        publicHooks[str](...args);
+        return m;
+      }
+
+      // Ensure the hook callback function is executed in the absence of a hook handler,
+      // so that execution may continue.
+      if (typeof args[1] === 'function') {
+        args[1]();
+      }
```

Comment:
Runs `args[1]` since the inputs to a public hook are the context and the callback function, in that order.